sheerwater_benchmarking/forecasts/fuxi.py

```diff
@@ -0,0 +1,230 @@
+#                    output_chunks={"lat": 721, "lon": 1440})
+#
+#    return ds
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['agg_days', 'prob_type'],
+           chunking={'lat': 121, 'lon': 240, 'lead_time': 14, 'time': 2, 'member': 51})
+def fuxi_rolled(start_time, end_time, agg_days=7, prob_type='probabilistic'):
+    """Roll and aggregate the FuXi data."""
+    ds = fuxi_raw(start_time, end_time)
+
+    # If deterministic average across the members
+    if prob_type == 'deterministic':
+        ds = ds.mean(dim='member')
+        ds = ds.assign_attrs(prob_type="deterministic")
+    else:
+        ds = ds.assign_attrs(prob_type="ensemble")
+
+    ds = roll_and_agg(ds, agg=agg_days, agg_col="lead_time", agg_fn="mean")
+
+    return ds
+
+@dask_remote
+@cacheable(data_type='array',
+           timeseries='time',
+           cache=False,
+           cache_args=['variable', 'lead', 'prob_type', 'grid', 'mask', 'region'])
+def fuxi(start_time, end_time, variable, lead, prob_type='deterministic',
+         grid='global1_5', mask='lsm', region="global"):
+    """Final fuxi forecast interface."""
+    if grid != 'global1_5':
+        raise NotImplementedError("Only 1.5 grid implemented for FuXi.")
+
+    """Standard format forecast data for daily forecasts."""
+    lead_params = {
+        "week1": ('weekly', 0),
+        "week2": ('weekly', 7),
+        "week3": ('weekly', 14),
+        "week4": ('weekly', 21),
+        "week5": ('weekly', 28),
+        "week6": ('weekly', 35),
+        "weeks12": ('biweekly', 0),
+        "weeks23": ('biweekly', 7),
+        "weeks34": ('biweekly', 14),
+        "weeks45": ('biweekly', 21),
+        "weeks56": ('biweekly', 28),
+    }
+    time_group, lead_offset_days = lead_params.get(lead, (None, None))
+    if time_group is None:
+        raise NotImplementedError(f"Lead {lead} not implemented for ECMWF debiased forecasts.")
+
+    # Convert start and end time to forecast start and end based on lead time
+    forecast_start = target_date_to_forecast_date(start_time, lead)
+    forecast_end = target_date_to_forecast_date(end_time, lead)
+
+    # Get the data with the right days
+    agg_days = lead_to_agg_days(lead)
+    ds = fuxi_rolled(forecast_start, forecast_end, prob_type=prob_type, agg_days=agg_days)
+    ds = lon_base_change(ds)
+
+
+    # Get the right variable
+    attrs = ds.attrs
+    ds = ds[variable].to_dataset()
+    ds = ds.assign_attrs(attrs)
+
+    # flip the lat dim
+    ds = ds.reindex(lat=list(reversed(ds.lat)))
+
+    # convert to mm
+    #if variable == 'precip':
+    #    ds['precip'] = ds['precip'] * 1000
+
+    # Convert from kelvin
+    if variable == 'tmp2m':
+        ds['tmp2m'] = ds['tmp2m'] - 273.15
+
+    # Get specific lead
+    lead_shift = np.timedelta64(lead_offset_days + 1, 'D')
+    ds = ds.sel(lead_time=lead_shift)
+
+    # Time shift - we want target date, instead of forecast date
+    ds = shift_forecast_date_to_target_date(ds, 'time', lead)
```

Comment:
Other than the lead labeling / interpretation, this looks good to me - it correctly converts time "2015-01-03" o time "2015-01-17" for the week3 forecast.